### PR TITLE
revert some delta request logic

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

https://github.com/istio/istio/pull/36269/files#r758456339

### CODE REVIEW

**Review by ramaraochavali — APPROVED** [review]

File: pilot/pkg/xds/ads.go [hunk 1]
```diff
@@ -380,7 +380,7 @@ func (s *DiscoveryServer) shouldRespond(con *Connection, request *discovery.Disc
 	// This can happen in two cases:
 	// 1. Envoy initially send request to Istiod
 	// 2. Envoy reconnect to Istiod i.e. Istiod does not have
-	// information about this typeUrl, but Envoy sends response nonce - either
+	// information about this typeUrl, but Envoy sendspilot/pkg/xds/delta.go response nonce - either
```
Comment (ramaraochavali) [inline pilot/pkg/xds/ads.go:383]: unnecessary change

Comment (ramaraochavali) [inline pilot/pkg/xds/ads.go:383]: unnecessary change

Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go:383]: not sure why it messed up, will clean

**Review by howardjohn — COMMENTED** [review]
lgtm, thanks!
